packages/resilience/src/retry.ts

```diff
@@ -0,0 +1,35 @@
+import * as asyncRetry from 'async-retry';
+import { Context, Middleware, MiddlewareIn, MiddlewareOut } from './middleware';
+
+const defaultRetryCount = 3;
+
+/**
+ * Helper method to build a retry middleware.
+ * @param retryCount - Number of retry attempts. Default value is 3.
+ * @returns The middleware adding a retry to the function.
+ */
+export function retry<ReturnType, ContextType extends Context>(
+  retryCount: number = defaultRetryCount
```

Comment:
[pp] I would prefer to put the aysnc-retry options here with a default setup. So that users can also adjust more then the count. On the other hand we are more simple doing that. @jjtang1985 do you have an opinion on that?